crates/wasm-mutate-stats/src/bin/wasm-mutate-stats.rs

```diff
@@ -0,0 +1,327 @@
+
+        loop {
+            match corpus.pop() {
+                Some((name, data)) => {
+                    println!("Wasm input {}", name);
+                    let mut wasm = data;
+
+                    let mut report = self.load_report(&name, wasm.clone()).unwrap();
+
+                    wasmmutate.seed(report.seed);
+                    let mut rng = SmallRng::seed_from_u64(report.seed);
+                    // mutate the initial wasm as many rotations are in the loaded report
+                    print!("Reaching previous state...");
+                    for _ in 0..report.rotations {
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                wasm = mutated;
+                            }
+                            Err(_) => { /* bypass */ }
+                        }
+                    }
+                    println!("Reached");
+                    while report.rotations <= ROTATIONS {
+                        report.rotations += 1;
+                        match wasmmutate.run_with_rng(&wasm, &mut rng) {
+                            Ok(mutated) => {
+                                let mut validator = wasmparser::Validator::new();
+                                match validator.validate_all(&mutated) {
+                                    Ok(_) => {
+                                        match self.compile(&mutated, &name, report.rotations) {
+                                            Ok(mhashes) => {
+                                                let hash = self.hash(mutated.clone());
+                                                wasm = mutated;
+                                                report.add_wasm_hash(hash);
+
+                                                for (idx, mhash) in mhashes.iter().enumerate() {
+                                                    report.add_obj_hash(mhash.clone(), idx);
+                                                }
+
+                                                if report.rotations % 100 == 0 {
+                                                    // Save report
+                                                    self.save_report(&report, &name);
+                                                    if self.print {
+                                                        report.print();
+                                                    }
+                                                }
+                                            }
+                                            Err(e) => report
+                                                .errors
+                                                .push((report.rotations as u64, format!("{}", e))),
+                                        }
+                                    }
+                                    // All generated Wasm should be valid
+                                    Err(e) => panic!("{}", e),
```

Comment:
Can you do `validator.validate_all(&mutated).expect("all generated Wasm should be valid")` instead of `match`ing? This way the code is a little less verbose and we can cut down on the deep indentation here.